x-pack/plugins/security_solution/public/common/lib/cell_actions/default_cell_actions.tsx

```diff
@@ -45,139 +43,167 @@ const useKibanaServices = () => {
+  const eventId = (header && get('_id' ?? '', ecs)) || '';
+  const rowData = useMemo(() => {
+    return {
+      data: data[pageRowIndex],
+      fieldName: columnId,
+    };
+  }, [pageRowIndex, columnId, data]);
 
-          const values = getMappedNonEcsValue({
-            data: data[pageRowIndex],
-            fieldName: columnId,
-          });
-
-          const value = parseValue(head(values));
-          return link && eventId && values && !isEmpty(value) ? (
-            <FormattedFieldValue
-              Component={Component}
-              contextId={`expanded-value-${columnId}-row-${pageRowIndex}-${timelineId}`}
-              eventId={eventId}
-              fieldFormat={header?.format || ''}
-              fieldName={columnId}
-              fieldType={header?.type || ''}
-              isButton={true}
-              isDraggable={false}
-              value={value}
-              truncate={false}
-              title={values.length > 1 ? `${link?.label}: ${value}` : link?.label}
-              linkValue={head(linkValues)}
-              onClick={closePopover}
-            />
-          ) : (
-            // data grid expects each cell action always return an element, it crashes if returns null
-            <></>
-          );
-        }
-      : EmptyComponent;
-  },
-];
+  const values = useGetMappedNonEcsValue(rowData);
+  const value = parseValue(head(values));
+  const title = values && values.length > 1 ? `${link?.label}: ${value}` : link?.label;
+  // if linkField is defined but link values is empty, it's possible we are trying to look for a column definition for an old event set
+  if (linkField !== undefined && linkValues.length === 0 && values !== undefined) {
+    const normalizedLinkValue = getField(ecs, linkField);
+    const normalizedLinkField = getFieldKey(ecs, linkField);
+    const normalizedColumnId = getFieldKey(ecs, columnId);
+    const normalizedLink = getLinkColumnDefinition(
+      normalizedColumnId,
+      header?.type,
+      normalizedLinkField
+    );
+    return {
+      pageRowIndex,
+      link: normalizedLink,
+      eventId,
+      fieldFormat: header?.format || '',
+      fieldName: normalizedColumnId,
+      fieldType: header?.type || '',
+      value: parseValue(head(normalizedColumnId)),
+      values,
+      title,
+      linkValue: head<string>(normalizedLinkValue),
+    };
+  } else {
+    return {
+      pageRowIndex,
+      link,
+      eventId,
+      fieldFormat: header?.format || '',
+      fieldName: columnId,
+      fieldType: header?.type || '',
+      value,
+      values,
+      title,
+      linkValue: head<string>(linkValues),
+    };
+  }
+};
 
 export const cellActions: TGridCellAction[] = [
   ({ data, pageSize }: { data: TimelineNonEcsData[][]; pageSize: number }) =>
-    ({ rowIndex, columnId, Component }: EuiDataGridColumnCellActionProps) => {
+    function FilterFor({ rowIndex, columnId, Component }: EuiDataGridColumnCellActionProps) {
```

Comment:
at some point the linter figured out that these functions that took props and returned jsx were indeed components, these functions are so that each component has a display name.